### PR TITLE
Fix: Native Compile

### PR DESCRIPTION
**Pull request by sgammon**
Ready for review [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=elide-dev&utm_campaign=badge_info)

## Summary

Fixes for native builds in the `cli` and `embedded` modules.

- [x] Fixes for building against GraalVM CE
- [x] Fixes for native targets
- [x] Re-enable WASM tests in CI

## Summary by CodeRabbit

- **New Features**
    - Added a new public static method to support WebAssembly in the GraalVM JavaScript class.
    - Added new tests for JavaScript and Python bindings in the context of GraalVM.
- **Bug Fixes**
    - Commented out sections related to installing runtime plugins, reflecting a shift in plugin management strategy.
- **Documentation**
    - Extended `.gitignore` to include `.codebase/` and `docs/reports/`.
- **Refactor**
    - Transitioned from GraalVM Enterprise to Community edition.
    - Streamlined GraalVM configuration and dependency management.
    - Adjusted CLI and embedded package configurations for improved performance and compatibility.
    - Updated JavaScript configuration for better feature support and compliance.
- **Chores**
    - Cleaned up workflow tests and adjusted build configurations for better efficiency and clarity.

### CODE REVIEW

File: packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt [hunk 1]
```diff
@@ -1546,40 +1546,40 @@ import elide.tool.project.ProjectManager
...
-        SCALA -> logging.warn("Scala runtime plugin is not yet implemented")
+//        PYTHON -> install(elide.runtime.plugins.python.Python) {
+//          logging.debug("Configuring Python VM")
+//          installIntrinsics(intrinsics, GraalVMGuest.PYTHON, versionProp)
+//        }
+
+//        RUBY -> install(elide.runtime.plugins.ruby.Ruby) {
+//          logging.debug("Configuring Ruby VM")
+//          installIntrinsics(intrinsics, GraalVMGuest.RUBY, versionProp)
+//        }
```
Comment (sgammon) [inline packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt:1557]: will refactor this in next PR

File: packages/embedded/build.gradle.kts [hunk 2]
```diff
@@ -117,7 +117,7 @@ val jvmCompileArgs = listOfNotNull(
   "--add-modules=jdk.incubator.vector",
   "-Delide.embedded.headerPath=$nativeHeaderPath",
   "--enable-native-access=" + listOfNotNull(
-    "ALL-UNNAMED",
+//    "ALL-UNNAMED",
```
Comment (sgammon) [inline packages/embedded/build.gradle.kts:120]: needs java 22

File: packages/cli/build.gradle.kts [hunk 3]
```diff
@@ -242,7 +229,7 @@ kapt {
 kotlin {
   target.compilations.all {
     kotlinOptions {
-      allWarningsAsErrors = true
+      allWarningsAsErrors = false
```
[the remark this answers is not in the document]
Comment (sgammon) [reply packages/cli/build.gradle.kts:232]: stfu nerd

**Comment by darvld** [comment]
@sgammon heads up looks like files under `docs/reports` did make it into Git, probably were added before the `.gitignore` was updated

File: packages/cli/build.gradle.kts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sgammon) [reply packages/cli/build.gradle.kts:232]: stfu nerd

**Review by darvld — CHANGES_REQUESTED** [review]
looks good excluding the previous comment

**Review by darvld — APPROVED** [review]

**Pull request merged**